src/IOSystem.h

```diff
@@ -50,14 +50,17 @@ class IOSystem {
     /**
      * Return a new WriteStream
      */
-    std::unique_ptr<WriteStream> getWriter(const std::vector<bool>& symbolMask,
-            const SymbolTable& symbolTable, const IODirectives& ioDirectives, const bool provenance) const {
+    std::unique_ptr<WriteStream> getWriter(const std::vector<char>& kindMask, const SymbolTable& symbolTable,
+            const std::vector<int>& recordArityMask, const RecordTable& recordTable,
```

Comment:
Can this be relied on to be an int or is there a better defined type to use?

Reply:
This is not set in stone and will likely change with the implementation of multi-level record printing, possibly to RamDomain or less, as arities might need to be stored as a field in record tuples.